fdbserver/storageserver.actor.cpp

```diff
@@ -9776,6 +9783,12 @@ ACTOR Future<Void> updateStorage(StorageServer* data) {
 				break;
 		}
 
+		// Allow data fetch to use an additional bytesLeft but don't penalize fetch budget if bytesLeft is negative
+		if (bytesLeft > 0) {
+			data->fetchKeysBytesBudget += bytesLeft;
+			data->fetchKeysBudgetUsed.set(data->fetchKeysBytesBudget <= 0);
```

Comment:
you need to wait for either durableDelay or data->fetchKeysBudgetUsed.onChange() here to give fetch keys a chance to write more KVs to redwood